### PR TITLE
XML Comments and Feedback from 1-on-1 with Nate

### PR DESCRIPTION
**Pull request by ZestyBread**
Couple of key things:
- Decided that API wise, having the keyword "Add" made it more clear that we are adding a rule rather than rewriting the options itself (or redirecting).
- UrlRewrite and ModRewrite are a bit unclear, so we added "Apache" and "IIS" to it.

### CODE REVIEW

**Comment by ZestyBread** [comment]
cc @Tratcher @natemcmaster

**Comment by ZestyBread** [comment]
@muratg for naming.

File: src/Microsoft.AspNetCore.Rewrite/ApacheModRewriteOptionsExtensions.cs [hunk 1]
```diff
@@ -8,15 +8,18 @@
 
 namespace Microsoft.AspNetCore.Rewrite
 {
-    public static class ModRewriteOptionsExtensions
+    /// <summary>
+    /// Apache Mod rewrite extensions on top of the <see cref="RewriteOptions"/>
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/ApacheModRewriteOptionsExtensions.cs]: nit: "Apache mod_rewrite"

File: src/Microsoft.AspNetCore.Rewrite/ApacheModRewriteOptionsExtensions.cs [hunk 2]
```diff
@@ -8,15 +8,18 @@
...
+    public static class ApacheModRewriteOptionsExtensions
     {
         /// <summary>
         /// Imports rules from a mod_rewrite file and adds the rules to current rules. 
         /// </summary>
         /// <param name="options">The Rewrite options.</param>
         /// <param name="hostingEnvironment">The Hosting Environment</param>
         /// <param name="filePath">The path to the file containing mod_rewrite rules.</param>
-        public static RewriteOptions ImportFromModRewrite(this RewriteOptions options, IHostingEnvironment hostingEnvironment, string filePath)
+        public static RewriteOptions AddApacheModRewrite(this RewriteOptions options, IHostingEnvironment hostingEnvironment, string filePath)
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/ApacheModRewriteOptionsExtensions.cs]: :+1:

File: src/Microsoft.AspNetCore.Rewrite/Internal/UrlActions/ChangeEnvironmentPreAction.cs [hunk 3]
```diff
@@ -4,17 +4,17 @@
 using System;
 using Microsoft.AspNetCore.Http;
 
-namespace Microsoft.AspNetCore.Rewrite.Internal.PreActions
+namespace Microsoft.AspNetCore.Rewrite.Internal.UrlActions
 {
-    public class ChangeEnvironmentPreAction : PreAction
+    public class ChangeEnvironmentPreAction : UrlAction
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Internal/UrlActions/ChangeEnvironmentPreAction.cs]: Can you also rename the class "ChangeEnvironmentPreAction" => "ChangeEnvironmentAction"?

File: src/Microsoft.AspNetCore.Rewrite/Internal/UrlActions/ChangeCookiePreAction.cs [hunk 4]
```diff
@@ -4,17 +4,17 @@
 using System;
 using Microsoft.AspNetCore.Http;
 
-namespace Microsoft.AspNetCore.Rewrite.Internal.PreActions
+namespace Microsoft.AspNetCore.Rewrite.Internal.UrlActions
 {
-    public class ChangeCookiePreAction : PreAction
+    public class ChangeCookiePreAction : UrlAction
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Internal/UrlActions/ChangeCookiePreAction.cs]: "ChangeCookiePreAction" => "ChangeCookieAction"

File: src/Microsoft.AspNetCore.Rewrite/RuleTermination.cs [hunk 5]
```diff
@@ -3,10 +3,16 @@
 
 namespace Microsoft.AspNetCore.Rewrite
 {
+    /// <summary>
+    /// An enum representing the result of a rule.
+    /// </summary>
     public enum RuleTermination
     {
+        // Default value, continue applying rules
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/RuleTermination.cs]: You can use the xmldoc format on enum values too. `  /// <summary></summary>`

File: src/Microsoft.AspNetCore.Rewrite/Rule.cs [hunk 6]
```diff
@@ -3,10 +3,16 @@
...
-    // make this public and doc comements
-    // caller must set the context.Results field appropriately in rule.
+    /// <summary>
+    /// Represents an abstract rule.
+    /// </summary>
     public abstract class Rule
     {
+        /// <summary>
+        /// Applies the rule.
+        /// Note that it is the responsibility of ApplyRule to set the context.Result appropriately.
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Rule.cs]: Suggested change of wording: "Implementations of ApplyRule should set a value for RewriteContext.Result."

**Comment by natemcmaster** [comment]
:watch:

**Comment by ZestyBread** [comment]
:up: :date:

File: src/Microsoft.AspNetCore.Rewrite/Internal/UrlActions/ChangeEnvironmentAction.cs [hunk 7]
```diff
@@ -4,17 +4,17 @@
 using System;
 using Microsoft.AspNetCore.Http;
 
-namespace Microsoft.AspNetCore.Rewrite.Internal.PreActions
+namespace Microsoft.AspNetCore.Rewrite.Internal.UrlActions
 {
-    public class ChangeEnvironmentPreAction : PreAction
+    public class ChangeEnvironmentPreAction : UrlAction
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Internal/UrlActions/ChangeEnvironmentAction.cs]: Nit: sync class name to filename. => "ChangeEnvironmentAction"

**Comment by natemcmaster** [comment]
LGTM.

**Comment by ZestyBread** [comment]
I found a bug for flags, didn't check for the OR flag.

**Comment by muratg** [comment]
:shipit: from the naming point of view. @natemcmaster or @Tratcher should give the final sign-off.

File: src/Microsoft.AspNetCore.Rewrite/Internal/CodeRules/RewriteRule.cs [hunk 8]
```diff
@@ -81,6 +80,7 @@ public override void ApplyRule(RewriteContext context)
                         else
                         {
                             request.Path = PathString.FromUriComponent(ForwardSlash + result);
+
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Internal/CodeRules/RewriteRule.cs]: Nit: delete empty line

**Comment by natemcmaster** [comment]
:shipit:

**Pull request merged**